src/core/mempool.rs

```diff
@@ -255,31 +256,36 @@ impl<'a> MemPoolTx<'a> {
         self.tx.commit().map_err(db_error::SqliteError)
     }
 
-    fn is_block_in_fork(
+    fn are_blocks_in_same_fork(
         &mut self,
         chainstate: &mut StacksChainState,
-        check_consensus_hash: &ConsensusHash,
-        check_stacks_block: &BlockHeaderHash,
-        cur_consensus_hash: &ConsensusHash,
-        cur_stacks_block: &BlockHeaderHash,
+        first_consensus_hash: &ConsensusHash,
+        first_stacks_block: &BlockHeaderHash,
+        second_consensus_hash: &ConsensusHash,
+        second_stacks_block: &BlockHeaderHash,
     ) -> Result<bool, db_error> {
-        let admitter_block =
-            StacksBlockHeader::make_index_block_hash(cur_consensus_hash, cur_stacks_block);
-        let index_block =
-            StacksBlockHeader::make_index_block_hash(check_consensus_hash, check_stacks_block);
+        let first_block =
+            StacksBlockHeader::make_index_block_hash(first_consensus_hash, first_stacks_block);
+        let second_block =
+            StacksBlockHeader::make_index_block_hash(second_consensus_hash, second_stacks_block);
         // short circuit equality
-        if admitter_block == index_block {
+        if second_block == first_block {
             return Ok(true);
         }
 
-        let height_result = chainstate
-            .with_clarity_marf(|marf| marf.get_block_height_of(&index_block, &admitter_block));
-        match height_result {
-            Ok(x) => {
-                eprintln!("{} from {} => {:?}", &index_block, &admitter_block, x);
-                Ok(x.is_some())
-            }
-            Err(x) => Err(db_error::IndexError(x)),
+        let height_of_first_with_second_tip = chainstate
```

Comment:
I'd prefer that you didn't use `with_clarity_marf()` for this, and instead used `get_index_tip_ancestor_conn()` and used that instead.  The reason is that `with_clarity_marf()` requires a _mutable_ reference to a `StacksChainState` instance, whereas `get_index_tip_ancestor_conn()` does not.  We're trying to keep the number of places where a `&mut StacksChainState` is needed outside of the relayer thread to a minimum, and this code path is executed through the networking thread.